static/js/Autocomplete.jsx

```diff
@@ -94,40 +120,55 @@ const SearchInputBox = ({getInputProps, suggestions, highlightedIndex,
      getInputProps().onChange({ target: { value: '' } });
   }
    const _submitSearch = (query) => {
-    Sefaria.getName(query)
-      .then(d => {
-        const repairedCaseVariant = Sefaria.repairCaseVariant(query, d);
-        if (repairedCaseVariant !== query) {
-          _submitSearch(repairedCaseVariant);
-          return;
-        }
-        const repairedQuery = Sefaria.repairGershayimVariant(query, d);
-        if (repairedQuery !== query) {
-          _submitSearch(repairedQuery);
-          return;
-        }
+      getQueryObj(query).then(queryObj => {
```

Comment:
you can deconstruct the queryObj here instead of on line 125 and 126 like this `{type: queryType, id: queryId}`